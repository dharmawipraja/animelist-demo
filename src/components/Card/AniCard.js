import * as React from 'react';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { Button, CardActionArea, CardActions } from '@mui/material';

function AniCard({ item, onClick }) {
  const { title, coverImage } = item;

  return (
    <Card sx={{ maxWidth: 250 }}>
      <CardActionArea onClick={onClick}>
        <CardMedia
          component="img"
          height="300"
          image={coverImage.large}
          alt="green iguana"
        />
        <CardContent>
          <Typography gutterBottom variant="h6" component="div" el noWrap>
            {title.romaji}
          </Typography>
        </CardContent>
      </CardActionArea>
      <CardActions>
        <Button size="small" color="primary">
          Add To Collections
        </Button>
      </CardActions>
    </Card>
  );
}

export default AniCard;
