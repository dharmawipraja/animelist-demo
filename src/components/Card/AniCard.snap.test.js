import renderer from 'react-test-renderer';
import AniCard from './AniCard';

let props = {
  item: {
    id: 1,
    title: {
      romaji: 'Test Title'
    },
    description: 'This is description',
    bannerCover: 'http://banner-cover.mock',
    coverImage: {
      large: 'http://banner-cover.mock',
      color: 'white'
    }
  },
  notitle: false,
  onCardClick: jest.fn(), 
  removable: false,
  onButtonClick: jest.fn(),
  isCollection: false
};

describe('AniCard', () => {
  it('should renders AniCard component', () => {
    const tree = renderer
      .create(<AniCard {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders AniCard component when it is on collections', () => {
    props = {
      ...props,
      isCollection: true
    };

    const tree = renderer
      .create(<AniCard {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should renders AniCard component when no title', () => {
    props = {
      ...props,
      notitle: true
    };

    const tree = renderer
      .create(<AniCard {...props} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
