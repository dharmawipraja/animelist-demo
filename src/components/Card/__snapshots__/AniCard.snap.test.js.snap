// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AniCard should renders AniCard component 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-103chds-MuiPaper-root-MuiCard-root"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="Test Title"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="300"
      src="http://banner-cover.mock"
    />
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-noWrap css-d5bee8-MuiTypography-root"
        el="true"
      >
        Test Title
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ikpx4h-MuiTypography-root"
      >
        Add To Collections
      </p>
      <svg
        color="#5c728a"
        fill="transparent"
        height={18}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": "#5c728a",
          }
        }
        viewBox="0 0 24 24"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`AniCard should renders AniCard component when it is on collections 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-103chds-MuiPaper-root-MuiCard-root"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="Test Title"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="300"
      src="http://banner-cover.mock"
    />
    <div
      className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <div
        className="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom MuiTypography-noWrap css-d5bee8-MuiTypography-root"
        el="true"
      >
        Test Title
      </div>
    </div>
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ikpx4h-MuiTypography-root"
      >
        Add To Collections
      </p>
      <svg
        color="red"
        fill="red"
        height={18}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": "red",
          }
        }
        viewBox="0 0 24 24"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
        />
      </svg>
      <span
        className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`AniCard should renders AniCard component when no title 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-103chds-MuiPaper-root-MuiCard-root"
>
  <button
    className="MuiButtonBase-root MuiCardActionArea-root css-b6lsxj-MuiButtonBase-root-MuiCardActionArea-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[MockFunction]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <img
      alt="Test Title"
      className="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
      height="300"
      src="http://banner-cover.mock"
    />
    <span
      className="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </button>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall MuiButton-textSizeSmall MuiButtonBase-root  css-1knaqv7-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-ikpx4h-MuiTypography-root"
      >
        Add To Collections
      </p>
      <svg
        color="red"
        fill="red"
        height={18}
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        style={
          Object {
            "color": "red",
          }
        }
        viewBox="0 0 24 24"
        width={18}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M20.84 4.61a5.5 5.5 0 0 0-7.78 0L12 5.67l-1.06-1.06a5.5 5.5 0 0 0-7.78 7.78l1.06 1.06L12 21.23l7.78-7.78 1.06-1.06a5.5 5.5 0 0 0 0-7.78z"
        />
      </svg>
    </button>
  </div>
</div>
`;
